doc/lightning-pay.7.md

```diff
@@ -43,6 +43,11 @@ creates an absolute limit on what fee we will pay.  This allows you to
 implement your own heuristics rather than the primitive ones used
 here.
 
+*description* is only required for bolt11 invoices which do not
+contain a description themselves, but contain a description hash.
+*description* is then checked against the hash inside the invoice
+before it will be paid.
```

Comment:
This requirement is incredibly restrictive: it limits us to invoices with `description_hash` where the `description` itself is representable as a raw JSON value. Anything like a PDF, which is something that people have been asking for, is essentially impossible to do with this, unless we allows a hex-encoded value to be provided instead, which makes this cumbersome to use. I know I voted in favor of requiring the `description` but this version is too limiting imho

Reply:
Yes, the limits of json!

But we can add descriptionhex later for this case?

Reply:
I looked at this, we need a non-trivial amount of re-engineering to add this.  We cannot handle bolt11 descriptions which are not basically valid strings right now, and changing that would be an API change everywhere we print description.

So I'm leaving that for next commit.